pkg/metrics/controller_test.go

```diff
@@ -40,29 +113,32 @@ func TestControllerGameServerCount(t *testing.T) {
 	gs1.Status.State = v1alpha1.GameServerStateReady
 	c.gsWatch.Modify(gs1)
 
+	c.run(t)
 	c.sync()
 	c.collect()
-	report()
 
 	gs1 = gs1.DeepCopy()
 	gs1.Status.State = v1alpha1.GameServerStateShutdown
 	c.gsWatch.Modify(gs1)
 	c.gsWatch.Add(gameServerWithFleetAndState("", v1alpha1.GameServerStatePortAllocation))
 	c.gsWatch.Add(gameServerWithFleetAndState("", v1alpha1.GameServerStatePortAllocation))
 
+	c.run(t)
 	c.sync()
 	c.collect()
-	report()
-
-	assert.Nil(t, testutil.GatherAndCompare(registry, strings.NewReader(gsCountExpected), "agones_gameservers_count"))
+	reader.ReadAndExport(exporter)
+	err := verifyMetricData(exporter, "gameservers_count", []expectedMetricData{
+		{labels: []string{"test-fleet", "Ready"}, val: int64(0)},
+		{labels: []string{"test-fleet", "Shutdown"}, val: int64(1)},
+		{labels: []string{"none", "PortAllocation"}, val: int64(2)},
+	})
+	assert.Nil(t, err)
```

Comment:
If you keep returning an error, `assert.NoError` is better, since if there is an error, it returns the error's message.